test/tests/functional/pbs_soft_walltime.py

```diff
@@ -857,39 +857,44 @@ def test_soft_held(self):
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
 
         self.logger.info(
-            "Sleep to let soft_walltime get extended once")
+            "Sleep to let soft_walltime get extended")
         time.sleep(10)
+
         self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
-        self.server.expect(JOB, {'estimated.soft_walltime':
-                                 (MATCH_RE, '00:00:14|14')}, id=jid)
+
+        self.server.expect(JOB, {'estimated.soft_walltime': 7}, op=GT,
+                           id=jid)
+
+        # Save the soft_walltime before holding the job
+        jstat = self.server.status(JOB, id=jid,
+                                   attrib=['estimated.soft_walltime'])
+        est_soft_walltime = jstat[0]['estimated.soft_walltime']
 
         self.server.holdjob(jid, 'u')
         self.server.rerunjob(jid)
         self.server.expect(JOB, {'job_state': 'H'}, id=jid)
 
         self.logger.info(
-            "Sleep to verify that soft_walltime"
-            " doesn't change while job is held")
+            "Sleep to verify that soft_walltime: %s"
+            " doesn't change while job is held" % est_soft_walltime)
         time.sleep(10)
         self.server.expect(JOB, {'estimated.soft_walltime':
-                                 (MATCH_RE, '00:00:14|14')}, id=jid)
+                           est_soft_walltime}, id=jid)
 
         # release the job and look for the soft_walltime again
         self.server.rlsjob(jid, 'u')
         self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
         self.server.expect(JOB, {'job_state': 'R', 'estimated.soft_walltime':
-                                 (MATCH_RE, '00:00:14|14')},
-                           attrop=PTL_AND, id=jid)
+                                 est_soft_walltime}, attrop=PTL_AND, id=jid)
```

Comment:
I believe will fail in API mode, since est_soft_walltime could be `00:00:14`, while in API mode the estimated.soft_walltime could be `14`. I think you might need to parse the time in seconds and use it in the `MATCH_RE`.

Reply:
No realistically it doesn't.  I was unaware when I wrote the test that the server's decode function was what created the HH:MM:SS form.  I thought it reported the number of seconds and qstat was doing the conversion.  The HH:MM:SS form will be reported in either CLI or API mode.  The problem is consistency with the rest of the test plan.  I suggest either do the MATCH_RE as @vstumpf suggested, or update the entire test plan to just expect() on the HH:MM:SS form.

Reply:
@vstumpf , @bhroam , Thanks for the comments. I am making the changes to the entire test plan as I see that other tests have also been failing in some test bed machines due to similar race conditions.